### PR TITLE
Pass SSL arg to all requests in DruidOperator

### PR DESCRIPTION
**Pull request by danielbe11**
- We're currently only passing the `verify` kwarg in the first post request (the one assigning `req_index` var). We should be passing the arg to all the requests in the operator. 
- Note that `request` sets this True by default [ref](https://github.com/psf/requests/blob/main/src/requests/api.py#L39)
- For context, the original change was in #37673 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by dirrao — CHANGES_REQUESTED** [review]
LGTM. Can you add the test with verify is set to True?

**Comment by danielbe11** [comment]
Thanks for the review @dirrao, added the other test

File: airflow/providers/apache/druid/hooks/druid.py [hunk 1]
```diff
@@ -144,13 +144,15 @@ def submit_indexing_job(
 
         sec = 0
         while running:
-            req_status = requests.get(druid_task_status_url, auth=self.get_auth())
+            req_status = requests.get(druid_task_status_url, auth=self.get_auth(), verify=self.get_verify())
```
Comment (danielbe11) [inline airflow/providers/apache/druid/hooks/druid.py:147]: Note that the default is True in both the [requests library](https://github.com/psf/requests/blob/main/src/requests/api.py#L39-L41) and [the init](https://github.com/apache/airflow/blob/main/airflow/providers/apache/druid/operators/druid.py#L56) of the hook, meaning it's backwards compatible

**Comment by danielbe11** [comment]
@dirrao could you please approve the pr if you're happy with it?

**Review by eladkal — APPROVED** [review]

**Pull request merged**